libnavigation-core/src/main/java/com/mapbox/navigation/core/reroute/MapboxRerouteController.kt

```diff
@@ -151,8 +151,8 @@ internal class MapboxRerouteController @VisibleForTesting constructor(
                 val origin = relevantAlternative.routerOrigin.mapToSdkRouteOrigin()
 
                 state = RerouteState.RouteFetched(origin)
-                callback.onNewRoutes(newList, origin)
                 state = RerouteState.Idle
+                callback.onNewRoutes(newList, origin)
```

Comment:
Why is this needed?

Reply:
This is needed to make sure that we can request a new reroute from an existing reroute callback. I'll also move this to a separate PR.